### PR TITLE
Translation files updated to work with Django 1.4, footer translated

### PR DESCRIPTION
**Pull request by premik91**
Footer translated, documentation updated.

I remember reading in one of your comments in other requests, that we shouldn't use django-admin.py, but since manage.py was moved, frendo's "tutorial" didn't work anymore, and I found this working replacement. I also hope that this is not another of my multi-feature problems. I think all of this belongs to translation feature.

### CODE REVIEW

File: docs/translation.rst [hunk 1]
```diff
@@ -4,36 +4,37 @@ Translation of PiplMesh
...
-in Python files like: ``_("some string")``.
+To make your strings translatable edit them as follows:
+
+- in Django template: ``{% trans "Some string" %}``
+- in Python files: ``_("some string")``
    
 Creation of translation files
 -----------------------------
    
 .. warning:: It is important to run folowing commands from repository root directory!
```
Comment (mitar) [inline docs/translation.rst:15]: In fact this is not correct. The proper way is still to run things from project directory, but to run it as:

```
../manage.py makemessages -l <language code>
```

See two dots.

File: piplmesh/frontend/templates/footer.html [hunk 2]
```diff
@@ -2,26 +2,29 @@
 <div id="footer">
     <hr class="hidden" />
     <div id="footer_nav">
-        <a href="http://wlan-si.net/">wlan slovenija</a> | <a href="https://nodes.wlan-si.net/">node database</a> | <a href="http://dev.wlan-si.net/">development</a> | <a href="http://grow.wlan-si.net/">growing</a>
+        <a href="http://wlan-si.net/">{% trans "wlan slovenija" %}</a> |
```
Comment (mitar) [inline piplmesh/frontend/templates/footer.html]: _wlan slovenija_ is name, this is not translated.

File: piplmesh/frontend/templates/footer.html [hunk 3]
```diff
@@ -2,26 +2,29 @@
...
-                                <input type="hidden" value="{{ request.get_full_path }}" />
-                                <input type="hidden" name="language" value="{{ lang.0 }}" />
-                                <button type="submit" class="text_button">{{ lang_info.name_local }}</button>
-                            </div>                       
-                        </form>
-                    </li>
-                {% endif %}
+            {% get_language_info for lang.0 as lang_info %}
+            {% if lang.0 == LANGUAGE_CODE %}
+            <li class="language_selected">{{ lang_info.name_local }}</li>
```
Comment (mitar) [inline piplmesh/frontend/templates/footer.html]: Plese indent all in the same way as it was before (and block/tag, template or HTML, is indented).

Comment (mitar) [inline piplmesh/frontend/templates/footer.html]: I am not sure what have you changed here except indentation?

File: piplmesh/locale/sl/LC_MESSAGES/django.po [hunk 4]
```diff
@@ -5,8 +5,8 @@
 msgid ""
 msgstr ""
 "Project-Id-Version: piplmesh\n"
-"Report-Msgid-Bugs-To: https://github.com/mitar/PiplMesh\n"
-"POT-Creation-Date: 2012-03-25 20:05+0200\n"
+"Report-Msgid-Bugs-To: \n"
```
Comment (mitar) [inline piplmesh/locale/sl/LC_MESSAGES/django.po]: Please make sure URL is kept (officially it should be e-mail address). But update it to new location.

File: piplmesh/locale/sl/LC_MESSAGES/django.po [hunk 5]
```diff
@@ -16,103 +16,144 @@ msgstr ""
...
 
-#: settings.py:49
+#: settings.py:43
 msgid "English"
 msgstr "Angleščina"
 
-#: account/forms.py:14
-msgid "Male"
+#: account/fields.py:9
+#, fuzzy
```
Comment (mitar) [inline piplmesh/locale/sl/LC_MESSAGES/django.po]: Remove those comments when you are reviewing translations. Always check the file.

File: piplmesh/locale/sl/LC_MESSAGES/django.po [hunk 6]
```diff
@@ -16,103 +16,144 @@ msgstr ""
...
+msgid "male"
 msgstr "Moški"
```
Comment (mitar) [inline piplmesh/locale/sl/LC_MESSAGES/django.po]: Case is obviously wrong. Fix.

File: piplmesh/locale/sl/LC_MESSAGES/django.po [hunk 7]
```diff
@@ -16,103 +16,144 @@ msgstr ""
...
+#, fuzzy
+msgid "female"
 msgstr "Ženska"
 
-#: account/forms.py:34
-msgid "Username"
-msgstr "Uporabniško ime"
+#: account/form_fields.py:11
+msgid "Value is out of bounds."
+msgstr "Vrednost je izven dovoljenega."
```
Comment (mitar) [inline piplmesh/locale/sl/LC_MESSAGES/django.po]: _Vrednost je izven meja._

File: piplmesh/locale/sl/LC_MESSAGES/django.po [hunk 8]
```diff
@@ -16,103 +16,144 @@ msgstr ""
...
-#: account/forms.py:59
+#: account/forms.py:79
 msgid "A user with that username already exists."
 msgstr "Uporabnik s tem uporabniškim imenom že obstaja."
 
-#: account/templates/login.html:5 account/templates/login.html.py:14
+#: account/signals.py:11
+#, fuzzy
+msgid "You have been successfully logged in."
+msgstr "Odjava uspešna."
```
Comment (mitar) [inline piplmesh/locale/sl/LC_MESSAGES/django.po]: _Prijava uspešna._

File: piplmesh/locale/sl/LC_MESSAGES/django.po [hunk 9]
```diff
@@ -16,103 +16,144 @@ msgstr ""
...
 msgstr "Gesli se ne ujemata."
 
-#: account/forms.py:59
+#: account/forms.py:79
 msgid "A user with that username already exists."
 msgstr "Uporabnik s tem uporabniškim imenom že obstaja."
 
-#: account/templates/login.html:5 account/templates/login.html.py:14
+#: account/signals.py:11
+#, fuzzy
```
Comment (mitar) [inline piplmesh/locale/sl/LC_MESSAGES/django.po]: Same.

File: piplmesh/locale/sl/LC_MESSAGES/django.po [hunk 10]
```diff
@@ -16,103 +16,144 @@ msgstr ""
...
+msgstr "Slika uporabnika"
 
-#: frontend/templates/header.html:32
+#: frontend/templates/header.html:38
 msgid "Log out"
 msgstr "Odjava"
 
-#: frontend/templates/header.html:35
+#: frontend/templates/header.html:43
 msgid "Log in"
```
Comment (mitar) [inline piplmesh/locale/sl/LC_MESSAGES/django.po]: Fix this to be always and everywhere _Login_ (one word).

File: piplmesh/locale/sl/LC_MESSAGES/django.po [hunk 11]
```diff
@@ -16,103 +16,144 @@ msgstr ""
...
 
 #: frontend/templates/header.html:31
-msgid "Username:"
-msgstr "Uporabniško ime:"
+msgid "User picture"
+msgstr "Slika uporabnika"
 
-#: frontend/templates/header.html:32
+#: frontend/templates/header.html:38
 msgid "Log out"
```
Comment (mitar) [inline piplmesh/locale/sl/LC_MESSAGES/django.po]: One word.

File: piplmesh/locale/sl/LC_MESSAGES/django.po [hunk 12]
```diff
@@ -16,103 +16,144 @@ msgstr ""
...
+msgstr "Odjava uspešna."
+
+#: account/templates/registration/login.html:5
+#: account/templates/registration/login.html:17
 msgid "Login"
 msgstr "Prijavi se"
 
-#: account/templates/login.html:19
+#: account/templates/registration/login.html:25
 msgid "Log in with Facebook"
```
Comment (mitar) [inline piplmesh/locale/sl/LC_MESSAGES/django.po]: _Login_

File: piplmesh/locale/sl/LC_MESSAGES/django.po [hunk 13]
```diff
@@ -16,103 +16,144 @@ msgstr ""
...
-#: account/templates/registration.html:28
+#: account/templates/registration/registration.html:17
+msgid "Required"
+msgstr "Obvezno"
+
+#: account/templates/registration/registration.html:29
 msgid "Reset"
 msgstr "Ponastavi"
 
+#: frontend/templates/footer.html:5
```
Comment (mitar) [inline piplmesh/locale/sl/LC_MESSAGES/django.po:103]: Remove.

File: piplmesh/locale/sl/LC_MESSAGES/django.po [hunk 14]
```diff
@@ -16,103 +16,144 @@ msgstr ""
...
-#: frontend/templates/header.html:35
+#: frontend/templates/header.html:43
 msgid "Log in"
 msgstr "Prijava"
+
+#: frontend/templates/search.html:5
+msgid "Search"
+msgstr "Išči"
+
+#~ msgid "Male"
```
Comment (mitar) [inline piplmesh/locale/sl/LC_MESSAGES/django.po]: For those string, check why they are not needed anymore and if there is no error (like somebody forgot to mark strings as translatable), remove them.

File: docs/translation.rst (hunk 1 above)
Comment (premik91) [inline docs/translation.rst:15]: Oh I tried everything and went through django documentation and haven't found anything but django-admin.py. I didn't even try the simplest solution. Sorry for my sloppiness.

File: piplmesh/frontend/templates/footer.html (hunk 2 above)
Comment (premik91) [inline piplmesh/frontend/templates/footer.html]: I didn't translate it, I just (don't know why) added it as a translatable.

File: piplmesh/frontend/templates/footer.html (hunk 3 above)
Comment (premik91) [inline piplmesh/frontend/templates/footer.html]: I noticed now, PyCharm did the indetation, I changed only the 'footer_nav', added translation tags and moved them each to new line(I am just used to it, beacause it is easier to read, and as far as I know it does no harm). I will of course put it back and never do it again

File: piplmesh/locale/sl/LC_MESSAGES/django.po (hunk 5 above)
Comment (premik91) [inline piplmesh/locale/sl/LC_MESSAGES/django.po]: Yes I noticed them, but wasn't sure if I should delete them

File: piplmesh/frontend/templates/footer.html (hunk 3 above)
Comment (mitar) [inline piplmesh/frontend/templates/footer.html]: Splitting footer into multiple lines is probably a good thing, jut don't break indentation at the same time. :-)

File: piplmesh/locale/sl/LC_MESSAGES/django.po (hunk 5 above)
Comment (mitar) [inline piplmesh/locale/sl/LC_MESSAGES/django.po]: I am deleting it. As far I understand it just tells you that this was automatically derived from some similar translation. And that you should check it. But you can check what Internet says about this.

**Comment by premik91** [comment]
I am very sorry for my sloppiness and your troubles yet again(something here brings the worst in me). The thing was, I just wanted to update and add things, so I wasn't paying enough attention to what has changed since the last translation(and many things did). I have also searched for the missing translations and haven't been able to find them in the code (but they are certainly there on registration forms):

#~ msgid "Username"
#~ msgstr "Uporabniško ime"

#~ msgid "Password"
#~ msgstr "Geslo"

#~ msgid "Repeat password"
#~ msgstr "Ponovi geslo"

#~ msgid "Username:"
#~ msgstr "Uporabniško ime:"

#~ msgid "Password:"
#~ msgstr "Geslo:"

**Comment by mitar** [comment]
> (something here brings the worst in me).

What, you have a bottle of vine next to you?

**Comment by premik91** [comment]
Nope, just a latop and reading what stupid mistakes I did. No bottle needed, it is sad enough :)

**Comment by mitar** [comment]
> I have also searched for the missing translations and haven't been able to find them in the code (but they are certainly there on registration forms):

Don't worry about those, they are not in our code, but official Django distribution and are already translated there. So you can remove them.

**Comment by premik91** [comment]
Fixed

**Comment by mitar** [comment]
Good work and great self-initiative! Now we have translations again current.

**Pull request merged**